### ISSUE TITLE
Clang's Large Output Makes Completion Slow

### ISSUE THREAD

**Issue by trishume** [company-mode/company-mode#225]
**TL;DR:** `company-clang` could be sped up by pre-filtering clang's output with `grep prefix`.

When company-clang calls clang for completions, if there is not enough context clang will output a massive list of every single identifier in the global scope. For most projects that use any sort of library, this can be thousands and thousands of symbols (I counted 40k for an IOS project, even when it couldn't load the UI library which is the biggest one).

The thing is it only takes clang a fraction of a second to output all these lines, those 40k lines only took 0.2s to pipe to `wc -l` in my shell. However, emacs takes a LONG time to read them all into a buffer. For me company-clang never manages to activate in the global scope because it takes up to a minute to read the thousands of lines (sometimes even a megabyte of text) into it's `* clang-output *` buffer.

This could be dramatically improved if some sort of pre-filtering was used. Currently `company-clang` doesn't take advantage of the prefix. Those 40k lines contain thousands of internal things starting with `__` even if the current prefix is `prin`. If the output could for example be piped through `grep $prefix` before being put in the buffer, it would dramatically increase performance.

Maybe company could include a shell script that only is used on Posix systems that pipes clang's output through grep, or perhaps there is some way to make reading the output into emacs much faster.
